core/lensHelpers.ml

```diff
@@ -0,0 +1,302 @@
+open Types
+open Utility
+open LensUtility
```

Comment:
What is the difference between LensHelpers and LensUtility?  It sounds like LensHelpers contains the actual implementation of the relational lens operations (the core logic) so that LensHelpersClassic is the old version, while the other "helpers" are components used by the core functionality that work the same in either version.

Reply:
`LensHelpers` contains some of the core logic, while `LensUtility` is currently just datatypes that are standalone. `Value` and `Types` depend on `LensUtility`, while `LensHelpers` depends on `Value` and `Types`. `LensHelpersClassic` is the non-incremental implementation, while `LensHelpersCorrect` is the incremental one and both mainly just contain the put operations.